datacube_stats/main.py

```diff
@@ -940,5 +965,168 @@ def __call__(self, product, time, group_by):
+                 {k: v for k, v in self.feature.items() if "WY" in k.upper()}
+            years = "_".join(x for x in [v for k, v in self.filter_product['year'].items()])
+            nlon = self.filter_product['args']['name'].upper() + '_' + str(Id)
+            self.tile_index = (nlon, years)
+            self.filter_time = get_hydrological_months(self.filter_product)
+        elif self.filter_product.get('method') == 'by_tide_height':
+            lon = self.feature['lon']
+            lat = self.feature['lat']
+            nlon = self.filter_product['args']['name'].upper() + '_' + str(Id) + '_' + str(lon)
+            nlat = str(lat) + '_PER_' + str(self.filter_product['args']['tide_percent'])
+            # get all relevant date time lists
+            if "item" in self.filter_product['args']['name']:
+                self.tile_index = (nlon, nlat)
+                self.filter_time = range_tidal_data(all_dates, self.filter_product, lon, lat)
+            else:
+                all_list, list_low, list_high, ebb_flow = \
+                    extract_otps_computed_data(all_dates, date_ranges,
+                                               self.filter_product['args']['tide_percent'], lon, lat)
+                self.tile_index = (nlon, nlat)
+                self.filter_time = list_low if self.filter_product['args']['name'] == 'low' else list_high
+                # filter out dates as per sub classification of ebb flow
+                if self.filter_product.get('sub_class'):
+                    list_low, list_high, ebb_flow = filter_sub_class(self.filter_product, list_low, list_high, ebb_flow)
+                    _LOG.info("SUB class dates extracted %s for list low %s  and for list high %s",
+                              self.filter_product['sub_class'], list_low, list_high)
+                    nlon = self.filter_product['args']['sub_class'].upper() + "_" + \
+                        + self.filter_product['args']['name'].upper()
+                    self.tile_index = (nlon, lat)
+                    self.filter_time = list_low if self.filter_product['name'] == 'low' else list_high
+                    key = set(e[0] for e in self.filter_time)
+                    ebb_flow_class = [ff for ff in eval(ebb_flow) if ff[0] in key]
+                    # dynamically store ebb flow information
+                    self.filter_product['args']['ebb_flow'] = {'ebb_flow': ebb_flow_class}
+                    _LOG.info('\nCreated EBB FLOW for feature %d , length %d, time period: %s \t%s',
+                              Id, len(ebb_flow_class), date_ranges, str(ebb_flow_class))
+                _LOG.info('\n %s DATE LIST for feature %d length %d, time period: %s \t%s',
+                          self.filter_product['args']['name'].upper(), str(self.filter_product['args']['feature_id']),
+                          len(self.filter_time), date_ranges, str(self.filter_time))
+        for time_period in date_ranges:
```

Comment:
This code below this line looks very similar (or identical) to code elsewhere in this file, is it possible to share most of the implementation?